packages/alchemy/src/client/rpcClient.ts

```diff
@@ -17,16 +24,27 @@ export const createAlchemyPublicRpcClient = ({
       ? `${chain.rpcUrls.alchemy.http[0]}/${connectionConfig.apiKey ?? ""}`
       : connectionConfig.rpcUrl;
 
+  fetchOptions.headers = {
+    ...fetchOptions.headers,
+    "Alchemy-AA-Sdk-Version": VERSION,
+  };
+
+  if (connectionConfig.jwt != null) {
+    fetchOptions.headers = {
+      ...fetchOptions.headers,
+      Authorization: `Bearer ${connectionConfig.jwt}`,
+    };
+  }
+
   return createBundlerClient({
     chain: chain,
-    transport: http(rpcUrl, {
-      ...(connectionConfig.jwt != null && {
-        fetchOptions: {
-          headers: {
-            Authorization: `Bearer ${connectionConfig.jwt}`,
-          },
-        },
-      }),
-    }),
-  });
+    transport: http(rpcUrl, { fetchOptions }),
+  }).extend(() => ({
+    updateHeaders(newHeaders: HeadersInit) {
+      fetchOptions.headers = {
+        ...fetchOptions.headers,
+        ...newHeaders,
+      };
+    },
```

Comment:
Like other comment, `HeadersInit` could be types that won't spread correctly here.